pandas/core/arrays/integer.py

```diff
@@ -304,9 +305,18 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
         return integer_array(scalars, dtype=dtype, copy=copy)
 
     @classmethod
-    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
-        scalars = to_numeric(strings, errors="raise")
-        return cls._from_sequence(scalars, dtype, copy)
+    def _from_sequence_of_strings(cls,
+                                  strings: Sequence[str],
+                                  dtype: Dtype = None,
+                                  copy: bool = False) -> 'IntegerArray':
+        # Mask the NA location before sending to to_numeric to prevent
+        # undesirable cast to float which may lose precision
+        mask = isna(strings)
+        masked_strings = np.where(mask, 0, strings)
+
```

Comment:
i think we can do this instead in to_numeric itself

Reply:
Was thinking that too but the downside I think is we'd have to return both an array of values and a mask from that method otherwise we'd keep hitting the same issue (unless we have pd.to_numeric construct the Int64 array directly)

Reply:
> (unless we have pd.to_numeric construct the Int64 array directly)

yes that is a desirable outcome